File: string_1/NonStart.java
/**
 * Given 2 strings, return their concatenation, except omit the first char of
 * each. The strings will be at least length 1.
 * 
 * <ul>
 *  <li>nonStart("Hello", "There") → "ellohere"
 *  <li>nonStart("java", "code") → "avaode"
 *  <li>nonStart("shotl", "java") → "hotlava"
 * </ul>
 * 
 * @author LZ-FSDev
 * @see https://codingbat.com/prob/p143825
 * @since 17.0.1
 * @version 0.0.1
 */
public class NonStart {
    public static void main(String[] args) {
        System.out.println("nonStart(\"Hello\", \"There\") -> "
                             + nonStart("Hello", "There"));
        System.out.println("nonStart(\"java\", \"code\") -> "
                             + nonStart("java", "code") );
        System.out.println("nonStart(\"shotl\", \"java\") -> "
                             + nonStart("shotl", "java"));
    }

    /**
     * Given 2 strings, return their concatenation, except omit the first char
     * of each. The strings will be at least length 1.
     * 
     * @param a First string.
     * @param b Second string.
     * @return The 2 strings concatenated together without the first characters
     * of both strings.
     * @since 0.0.1
     */
    public static String nonStart(String a, String b) {
        return a.substring(1) + b.substring(1);
    }
}

File: string_1/NonStart.py
'''
Given 2 strings, return their concatenation, except omit the first char of
each. The strings will be at least length 1.
non_start('Hello', 'There') → 'ellohere'
non_start('java', 'code') → 'avaode'
non_start('shotl', 'java') → 'hotlava'
'''

def non_start(a, b):
    return a[1:] + b[1:]

print("non_start('Hello', 'There') -> " + str(non_start('Hello', 'There')))
print("non_start('java', 'code') -> " + str(non_start('java', 'code')))
print("non_start('shotl', 'java') -> " + str(non_start('shotl', 'java')))
print("non_start(' _', '_ _') -> " + str(non_start(' _', '_ _')))
print("non_start('wow', ' w') -> " + str(non_start('wow', ' w')))
